service/src/main/kotlin/fi/espoo/evaka/varda/VardaOrganizerChild.kt

```diff
@@ -0,0 +1,209 @@
+        if (rowsByChildAndOrganizer.isEmpty()) {
+            return@transaction createVardaChildWhenPersonExists(tx, client, evakaPersonId, vardaPersonOid, municipalOrganizerOid, organizerOid, isPaosChild, sourceSystem)
+        }
+
+        return@transaction rowsByChildAndOrganizer.first().vardaChildId
+    }
+}
+
+data class VardaChildOrganizerRow(
+    val evakaPersonId: UUID,
+    val vardaPersonOid: String,
+    val vardaChildId: Long,
+    val organizerOid: String
+)
+
+private fun getVardaOrganizationChildRows(
+    tx: Database.Transaction,
+    evakaPersonId: UUID,
+): List<VardaChildOrganizerRow> {
+    val sql = """
+        SELECT evaka_person_id, varda_person_oid, varda_child_id, organizer_oid
+        FROM varda_organizer_child
+        WHERE evaka_person_id = :evakaPersonId
+    """.trimIndent()
+
+    return tx.createQuery(sql)
+        .bind("evakaPersonId", evakaPersonId)
+        .mapTo<VardaChildOrganizerRow>()
+        .toList()
+}
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+data class VardaChildPayload(
+    @JsonProperty("henkilo_oid")
+    val personOid: String,
+    @JsonProperty("vakatoimija_oid")
+    val organizerOid: String,
+    @JsonProperty("lahdejarjestelma")
+    val sourceSystem: String
+)
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+data class VardaPaosChildPayload(
+    @JsonProperty("henkilo_oid")
+    val personOid: String,
+    @JsonProperty("oma_organisaatio_oid")
+    val organizerOid: String,
+    @JsonProperty("paos_organisaatio_oid")
+    val paosOrganizationOid: String,
+    @JsonProperty("lahdejarjestelma")
+    val sourceSystem: String
+)
+
+private fun getMunicipalOrganizerOid(tx: Database.Transaction): String {
+    return tx.createQuery("SELECT varda_organizer_oid FROM varda_organizer LIMIT 1") // one row table
+        .mapTo<String>()
+        .toList()
+        .first()
+}
+
+private fun createVardaPersonAndChild(
+    tx: Database.Transaction,
+    client: VardaClient,
+    evakaPersonId: UUID,
+    municipalOrganizerOid: String,
+    organizerOid: String,
+    isPaosChild: Boolean,
+    sourceSystem: String
+): Long {
+    val personPayload = getVardaPersonPayload(tx, evakaPersonId, organizerOid)
+
+    val vardaPersonOid = client.createPerson(personPayload)?.personOid ?: error("Couldn't create Varda person (nor child)")
+
+    return createVardaChildWhenPersonExists(
+        tx,
+        client,
+        evakaPersonId,
+        vardaPersonOid,
+        municipalOrganizerOid,
+        organizerOid,
+        isPaosChild,
+        sourceSystem
+    )
+}
+
+private fun createVardaChildWhenPersonExists(
+    tx: Database.Transaction,
+    client: VardaClient,
+    evakaPersonId: UUID,
+    vardaPersonOid: String,
+    municipalOrganizerOid: String,
+    organizerOid: String,
+    isPaosChild: Boolean,
+    sourceSystem: String
+): Long {
+    return if (isPaosChild) {
+        val vardaChildPayload = VardaPaosChildPayload(
```

Comment:
Term "VardaXRequest" is used elsewhere in the code if the request does not match the domain object

Reply:
Hmm I don't quite get it. Do you wish `VardaPaosChildPayload` to be renamed to something?

Reply:
I mean that the data class sent to varda has been named like "VardaChildRequest", so should this be "VardaPaosChildRequest"?

Reply:
Well they are not actually the request objects so I dunno, maybe not